### PR TITLE
Adding session file as input to schema subcommand and Updating schema subcommand documentation

### PR DESCRIPTION
**Pull request by VardhanThigle**
Fixes # 380773203

> It's a good idea to open an issue first for discussion.

- [ ] Tests pass - N/A
- [ ] Appropriate changes to README are included in PR - N/A

### CODE REVIEW

File: docs/cli/data.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ reference of the gCloud version of SMT, please refer [here](https://cloud.google
 ## REQUIRED FLAGS
 
      --session=SESSION
-        Specifies the file that you restore session state from.
+        Specifies the file that you restore session state from. This file can be generaed using the [schma](schema.md) sub command.
```
Comment (bharadwaj-aditya) [inline docs/cli/data.md:61]: typo: generated

File: docs/cli/schema.md [hunk 2]
```diff
@@ -8,9 +8,14 @@ nav_order: 1
...
-is then passed to the data subcommand to perform data migration while honoring the defined
-schema mapping. Spanner migration tool also generates Spanner schema which users can modify manually and use directly as well.
+This subcommand can be used to perform schema conversion and report on the quality of the conversion. 
+Based on the options discussed further, it helps with:
+1. Generate Report on quality of conversion.
+2. Generate the Spanner schema in Schema file, which could be manually modified and applied on spanner if required.
+3. Generate schema mapping file (`session.json`), which helps the data migration pipeline with the context how the source shcema maps to spanner schema. If required, the schema mapping file can be manually edited (either directly or with the help of SMT web UI). The modified session file can be passed back as **sessionFilePath** parameter to schema sub command if required.
+4. If you would like to perform the data migration via spanner migration tool, the session file needs be passed to the [data subcommand](data.md) as the **--session** parameter.
```
Comment (bharadwaj-aditya) [inline docs/cli/schema.md:16]: I'm not sure if this is required. Please verify

File: docs/cli/schema.md [hunk 3]
```diff
@@ -8,9 +8,14 @@ nav_order: 1
...
+5. In case you would like to use [sourcedb-to-spanner](https://github.com/GoogleCloudPlatform/DataflowTemplates/tree/94a7e9f08cadd4980201dfa8a7fc397970111021/v2/sourcedb-to-spanner) dataflow template for migrating the bulk data, you should updload the session file to GCS and pass the path to the same as **sessionFilePath**.
```
Comment (bharadwaj-aditya) [inline docs/cli/schema.md:17]: We should either mention all the templates or none.

File: cmd/schema.go [hunk 4]
```diff
@@ -123,12 +125,19 @@ func (cmd *SchemaCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interfa
 	schemaConversionStartTime := time.Now()
 	var conv *internal.Conv
 	convImpl := &conversion.ConvImpl{}
-	conv, err = convImpl.SchemaConv(cmd.project, sourceProfile, targetProfile, &ioHelper, &conversion.SchemaFromSourceImpl{})
-	if err != nil {
-		return subcommands.ExitFailure
+	if cmd.sessionJSON != "" {
```
Comment (bharadwaj-aditya) [inline cmd/schema.go:128]: can you add a test for this ?

File: docs/cli/data.md (hunk 1 above)
Comment (bharadwaj-aditya) [reply docs/cli/data.md:61]: We should also specify what happens if a session file and source profile are passed.

Comment (VardhanThigle) [reply docs/cli/data.md]: `generated`: Fixed in the recent push.
```
We should also specify what happens if a session file and source profile are passed.
```
Added in the docs for Schema sub command, and also added info logs around it.

File: docs/cli/schema.md (hunk 3 above)
Comment (VardhanThigle) [reply docs/cli/schema.md]: Removed this. Thanks for the clarification. This was added from the POV of needing clearer documentation for the specific use case.

File: docs/cli/schema.md (hunk 2 above)
Comment (VardhanThigle) [reply docs/cli/schema.md:16]: It's added to make the earlier version more clearer (line 12 of previous diff).

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Pull request merged**